src/cli/cli.cpp

```diff
@@ -708,6 +719,90 @@ void Interpreter::ProcessDiscover(int argc, char *argv[])
     AppendResult(error);
 }
 
+#if OPENTHREAD_ENABLE_DNS_CLIENT
+void Interpreter::ProcessDns(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    long port = OT_DNS_DEFAULT_SERVER_PORT;
+    Ip6::MessageInfo messageInfo;
+    otDnsQuery query;
+
+    if (argc && strcmp(argv[0], "resolve") == 0)
```

Comment:
What if `argc` is 0?  Right now it looks like it will simply return with no error or output.  Should this case return a `kThreadError_InvalidArgs`, since `argc == 0` has no defined behaviour?